webapp/src/components/enforceConnectedAccountModal.tsx

```diff
@@ -1,49 +1,73 @@
-import React, {useCallback, useState, useEffect} from 'react';
+import React, {useCallback, useState, useEffect, useMemo, useRef} from 'react';
 
-import Client from '../client';
-import {id as pluginId} from '../manifest';
+import Constants from 'constants/index';
+
+import usePluginApi from 'hooks/usePluginApi';
+import useApiRequestCompletionState from 'hooks/useApiRequestCompletionState';
 
 import './enforceConnectedAccountModal.css';
 
 export default function EnforceConnectedAccountModal() {
     const [open, setOpen] = useState(false);
     const [canSkip, setCanSkip] = useState(false);
     const [connecting, setConnecting] = useState(false);
-    const iconURL = `/plugins/${pluginId}/public/msteams-sync-icon.svg`;
+    const isInterval = useRef(false);
+    const {makeApiRequestWithCompletionStatus, getApiState} = usePluginApi();
 
     const skip = useCallback(() => {
         setOpen(false);
     }, []);
 
     const connectAccount = useCallback(() => {
-        Client.connect().then((result) => {
-            setConnecting(true);
-            window.open(result?.connectUrl, '_blank');
-        });
+        makeApiRequestWithCompletionStatus(Constants.pluginApiServiceConfigs.connect.apiServiceName);
     }, []);
 
     useEffect(() => {
-        Client.needsConnect().then((result) => {
-            setOpen(result.needsConnect);
-            setCanSkip(result.canSkip);
-        });
+        makeApiRequestWithCompletionStatus(Constants.pluginApiServiceConfigs.needsConnect.apiServiceName);
     }, []);
 
-    const checkConnected = useCallback(async () => {
-        const result = await Client.needsConnect();
-        if (!result.needsConnect) {
-            setOpen(false);
-            setConnecting(false);
-        }
+    const {data: needsConnectData} = useMemo(() => getApiState(Constants.pluginApiServiceConfigs.needsConnect.apiServiceName), [getApiState]);
```

Comment:
I have similar comments here about potentially not needing a `useMemo` around `getApiState` and being able to use `useEffect` instead of `useApiRequestCompletionState` for these cases.

The `useApiRequestCompletionState` one might be different in this case though since it's making repeated requests for that